node/src/requests/queue.rs

```diff
@@ -173,6 +173,8 @@ pub struct ComputationProgress<ChainRespondArgsType: ChainRespondArgs> {
     /// The computed response, if any. This is used to prevent unnecessary recomputation,
     /// if all that's needed is to submit the response to chain.
     pub computed_response: Option<ChainRespondArgsType>,
+    /// The leader selected during computation
+    pub selected_leader: Option<ParticipantId>,
```

Comment:
this was the actual trick to solve the issue, add the leader to the `ComputationProgress` struct

Reply:
Nice!